### DISCUSSION TITLE
PyArmor 7.7.4 does not generate/copy .key and .lic files

### DISCUSSION THREAD

**Discussion by Serob** [dashingsoft/pyarmor#1900] [Q&A] [answered]
I'm on Windows 10
Python 3.10.10 64-bit
Pyarmor 7.7.4
License: Enterprise

When I run `pyarmor -o my_script.py -O path_to_obfuscated_dir`, it creates the scripts without showing any error:

```
...
INFO     Use cached capsule path_to_my_pyarmor_capsule.zip
INFO     Search scripts mode: Normal
INFO     Read product key from capsule
INFO     Obfuscate module mode is 2
INFO     Obfuscate code mode is 1
INFO     Obfuscate string value is False
INFO     Wrap mode is 1
INFO     Restrict mode is 1
INFO     Advanced value is 0
INFO     Super mode is False
INFO     Super plus mode is not enabled
INFO     Generating runtime files to path\pytransform
INFO     Extract pytransform.key
INFO     Read default license from capsule
INFO     Copying Python310\lib\site-packages\pyarmor\platforms\windows\x86_64\_pytransform.dll
INFO     Patch library path\pytransform\_pytransform.dll
INFO     Patch library file OK
INFO     Copying Python310\lib\site-packages\pyarmor\pytransform.py
INFO     Rename it to pytransform/__init__.py
INFO     Generate runtime files OK

```
 but inside `pytransform` folder I can see only `__init__.py` and  `_pytransform.dll` files. 

Those obfuscated scripts run on another machine without any restriction, and if I create a `license.lic` file it does not have any effect
When I run `pyarmor --version` I can see that I'm an enterprise user

But when I try the same license with v5.8.9 and python 3.7 32-bit it works and creates both `license.lic` and `pytransform.key` files after obfuscation (and `license.lic` of course can be replaced by another one).

**Comment by jondy** [comment, marked as answer]
Please check Pyarmor 7 doc, and understand how to use outer license

**Reply by Serob** [reply]
Thanks. That helped!
Just pasting a URL for others to find things easily:
https://pyarmor.readthedocs.io/en/v7.7.0/advanced.html#how-to-use-outer-license-file
